controller/internal/enforcer/envoyauthorizer/envoyauthorizerenforcer.go

```diff
@@ -82,16 +97,207 @@ func (e *Enforcer) Secrets() (secrets.Secrets, func()) {
 }
 
 // Enforce starts enforcing policies for the given policy.PUInfo.
+// here we do the following:
+// 1. create a new PU always and instantiate a new apiAuth, as we want to be as stateless as possible.
+// 2. create a PUcontext as this will be used in auth code.
+// 3. If envoy servers are not present then create all 3 envoy servers.
+// 4. If the servers are already present under policy update then update the service certs.
 func (e *Enforcer) Enforce(contextID string, puInfo *policy.PUInfo) error {
 	e.Lock()
 	defer e.Unlock()
+
+	zap.L().Info("Enforce for the envoy remoteEnforcer for pu: ", zap.String("puID:", contextID))
```

Comment:
Make this a debug message, also this is not bound to be necessarily only be a remoteEnforcer